apis/v1alpha1/types.go

```diff
@@ -83,6 +164,39 @@ type EventSource struct {
 	Name      *string `json:"name,omitempty"`
 }
 
+// These are custom parameter to be used when the target is an API Gateway REST
+// APIs or EventBridge ApiDestinations. In the latter case, these are merged
+// with any InvocationParameters specified on the Connection, with any values
+// from the Connection taking precedence.
+type HTTPParameters struct {
+	HeaderParameters      map[string]*string `json:"headerParameters,omitempty"`
+	PathParameterValues   []*string          `json:"pathParameterValues,omitempty"`
+	QueryStringParameters map[string]*string `json:"queryStringParameters,omitempty"`
+}
+
+// Contains the parameters needed for you to provide custom input to a target
+// based on one or more pieces of data extracted from the event.
+type InputTransformer struct {
+	InputPathsMap map[string]*string `json:"inputPathsMap,omitempty"`
+	InputTemplate *string            `json:"inputTemplate,omitempty"`
+}
+
+// This object enables you to specify a JSON path to extract from the event
+// and use as the partition key for the Amazon Kinesis data stream, so that
+// you can control the shard to which the event goes. If you do not include
+// this parameter, the default is to use the eventId as the partition key.
+type KinesisParameters struct {
+	PartitionKeyPath *string `json:"partitionKeyPath,omitempty"`
+}
+
+// This structure specifies the network configuration for an ECS task.
+type NetworkConfiguration struct {
+	// This structure specifies the VPC subnets and security groups for the task,
+	// and whether a public IP address is to be used. This structure is relevant
+	// only for ECS tasks that use the awsvpc network mode.
+	AWSvpcConfiguration *AWSVPCConfiguration `json:"awsvpcConfiguration,omitempty"`
```

Comment:
The casing here seems off. We probably need to add an edge case to the initialisms list: https://github.com/aws-controllers-k8s/pkg/blob/main/names/names.go#L47

Reply:
This seems non-blocking then as it doesn't have user impact, right?

Reply:
It will, actually. The `json:"awsvpcConfiguration,omitempty"` part of the string implies that the name of the field within the spec is `awsvpcConfiguration` whereas we want it to be `awsVPCConfiguration`. If we publish the controller as GA with this casing, we can't go back and fix it or we will break backward compatibility. So we need to nip it in the bud now

Reply:
Ah, I thought this was about the type field naming and not JSON. If it's also about JSON then yes, would be breaking.

Reply:
Can you please help me patch this in `names.go` so I can regenerate the code once this is fixed? Unfortunately, don't have cycles to tackle this one now.

Reply:
@embano1 I'll get a patch up for those initialisms. Might I also suggest renaming awsVPCConfiguration to just `vpcConfiguration`, though? Seems like "aws" is a bit redundant ;)

Reply:
I'll follow your guidelines. CloudFormation is

```yaml
  AwsVpcConfiguration: 
    AwsVpcConfiguration
```

Not sure if customers expect the same field names in ACK?

Reply:
Agree with @embano1 here. If you're essentially copying your spec from another Cfn source it would be rough to figure out why it's not working because only one nested field has a different name

Reply:
Question: would copying also break if we tweak the field generator/initialism from `awsVpc` to `awsVPC`?

Reply:
I suppose so, too. But I think the users would be able to understand the need to change casing for all fields more easily than one arbitrary prefix deletion on the nested field. Maybe that's a bad assumption on my behalf though, and if we're going to make people change things about the spec then we may as well make it exactly as we want it